### ISSUE TITLE
Cannot start Razor server "Cannot get required symbol EVP_MD_size from libssl"

### ISSUE THREAD

**Issue by ZeroSkill1** [OmniSharp/omnisharp-vscode#5321]
## Environment data
`dotnet --info` output:
```
.NET SDK (gemäß "global.json"):
 Version:   6.0.301
 Commit:    43f9b18481

Laufzeitumgebung:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.301/

Host (useful for support):
  Version: 6.0.6
  Commit:  7cca709db2

.NET SDKs installed:
  3.1.420 [/usr/share/dotnet/sdk]
  5.0.408 [/usr/share/dotnet/sdk]
  6.0.301 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.26 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.6 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.26 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```
VS Code version:
```
Version: 1.69.2
Commit: 3b889b090b5ad5793f524b5d1d39fda662b96a2a
Date: 2022-07-18T16:14:10.636Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Linux x64 5.15.0-41-generic
```
C# Extension version: `v1.25.0`

## Steps to reproduce

1. Open any `.cshtml` file

## Expected  behavior
The Razor server should start.

## Actual behavior
The Razor server crashes repeatedly due to the following:
```
Cannot get required symbol EVP_MD_size from libssl
[Info  - 10:33:46] Connection to server got closed. Server will restart.
```
As a result, the server never starts.

**Comment by ZeroSkill1** [comment]
Launching VSCode by adding `CLR_OPENSSL_VERSION_OVERRIDE=1.1` to the launch command seems to have resolved this issue.

**Comment by muchirajunior** [comment]
which file are we supposed to add this

**Comment by elestedt** [comment]
@muchirajunior Add
```
    "terminal.integrated.env.linux": {
        "CLR_OPENSSL_VERSION_OVERRIDE": "1.1"
    }
```
to your settings.json, either for the project or for your user.

**Comment by muchirajunior** [comment]
still not working

**Comment by Dicusar94** [comment]
> Launching VSCode by adding `CLR_OPENSSL_VERSION_OVERRIDE=1.1` to the launch command seems to have resolved this issue.

Can you help us with more detail information how to fix the issue please.

**Comment by muchirajunior** [comment]
> > Launching VSCode by adding `CLR_OPENSSL_VERSION_OVERRIDE=1.1` to the launch command seems to have resolved this issue.
> 
> Can you help us with more detail information how to fix the issue please.

that does't solve this issue at all

**Comment by felipebueno** [comment]
I tried setting ``'CLR_OPENSSL_VERSION_OVERRIDE``` to 1.1 but it didin't work.

```
Cannot get required symbol EVP_MD_size from libssl
[Error - 11:33:11 AM] Connection to server got closed. Server will not be restarted.
[Error - 11:33:11 AM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:994214)
	at Object.dispose (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:1050936)
	at x.handleConnectionClosed (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:1051111)
	at x.handleConnectionClosed (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:1070276)
	at t (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:1049310)
	at n.invoke (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:979052)
	at r.fire (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:979782)
	at $ (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:986274)
	at n.invoke (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:979052)
	at r.fire (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:979782)
	at l.fireClose (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:997133)
	at Socket.<anonymous> (/home/usernamehere/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/dist/extension.js:2:997895)
	at Socket.emit (node:events:538:35)
	at Pipe.<anonymous> (node:net:687:12)
```

**Comment by fredriksknese** [comment]
Set CLR_OPENSSL_VERSION_OVERRIDE to 1.1 worked for me. Check the screenshot below to see how you can do it.

<img width="341" alt="image" src="https://user-images.githubusercontent.com/5236553/193462964-7b572cca-696f-4d2c-8934-a95065227815.png">

Since I'm using a user local version of dotnet I also had to change dotnet to its full path in the project-file .vscode/tasks.json for it to work as expected.
<img width="572" alt="image" src="https://user-images.githubusercontent.com/5236553/193463131-065200a5-e871-4cec-8064-0440b09847bb.png">

**Comment by muchirajunior** [comment]
I don't see how and where Set CLR_OPENSSL_VERSION_OVERRIDE to 1.1 is on the screen shot ..

**Comment by fredriksknese** [comment]
You don't see the screenshot with the command prompt?
Here is a direct-link to the above image: https://user-images.githubusercontent.com/5236553/193462964-7b572cca-696f-4d2c-8934-a95065227815.png

In text it reads:
```
C:\Users\fredr> set CLR_OPENSSL_VERSION_OVERRIDE=1.1
C:\Users\fredr> code
```

**Comment by abdollahkahne** [comment]
Thanks for every one for providing this fix. Just to review the solution. I first checked that LibSSL1.1 installed:
![image](https://user-images.githubusercontent.com/48277995/211844023-10a869f3-be8c-47ef-86c2-573758dd3760.png)
Then I open terminal and run the following commands:
![image](https://user-images.githubusercontent.com/48277995/211844304-051fa536-9612-4e65-80ce-7d5ff6970c23.png)
ٍ
echo is not necessary and just to check the value is set. It works for me (Ubuntu 22.04 upgraded from 20.04). I could not do this with `set`, so I used `export` instead to set shell environment variables.
